packages/cosmic-swingset/x/swingset/module.go

```diff
@@ -71,10 +71,12 @@ type AppModule struct {
 
 // NewAppModule creates a new AppModule Object
 func NewAppModule(k Keeper) AppModule {
-	return AppModule{
+	am := AppModule{
 		AppModuleBasic: AppModuleBasic{},
 		keeper:         k,
 	}
+	RegisterPortHandler("dibc", NewIBCChannelHandler(am))
```

Comment:
Global registration as a side-effect of construction is surprising. If the module can effectively only be constructed once, perhaps the keeper should be a global too and these should be assembled in an `init()`. Having a public function implies it could be called by a user, possibly more than once.